flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/table/DynamoDbDynamicSinkFactory.java

```diff
@@ -58,6 +58,17 @@ public DynamicTableSink createDynamicTableSink(Context context) {
                         .setDynamoDbClientProperties(
                                 dynamoDbConfiguration.getSinkClientProperties());
 
+        if (catalogTable.getResolvedSchema().getPrimaryKey().isPresent()) {
+            builder =
+                    builder.setPrimaryKeys(
+                            new HashSet<>(
```

Comment:
DynamoDB primary key is an ordered set of properties - partitionKey and then sortKey. Let's model it like that.

I would suggest using those as dedicated properties instead in the Sink model. We could add an extra validation here to ensure consistency with DDB schema.

Reply:
+1
It must be an ordered set/list of 1 or 2 elements.

Suggestion: if not too complicated, for a user perspective it would be clearer a no-nonsense passing a *PrimaryKey* object with two fields, *partitionKey* and *sortKey*, where *partitionKey*

Reply:
@nicusX - While having a dedicated `PrimaryKey` object with two fields _partitionKey_ and _sortKey_ might work for the DataStream API if I were to add it to the Sink model, I am not clear on how that would then translate to the Table API.  I was hoping to just use the fact that the Table API / SQL API already supports the concept of passing in a `PRIMARY KEY` for that.  

I don't think there's any reason here why we would need to separate the partition key and sort key for the purpose here of identifying the Primary Key - and in fact, there is also some naming collision then as well as the current Table API / SQL connector supports passing in a `PARTITIONED BY` clause, adding to potential confusion.

Reply:
Ack for not having an object with separate fields.
However, in this case the `primaryKey` (singular) should be an ordered set (e.g. a List) and not a Set. The order of the two fields is relevant: the first always being the *partitionKey*, in DDB parlance, and the second, if present, the *sortKey*